R/utils.R

```diff
@@ -250,11 +250,23 @@ simplify_string <- function(x) {
   tolower(gsub("\\s+", "", x))
 }
 
-
 url_format <- function(url) {
-
   url <- gsub("&", "&\n", url)
   sub("SERVICE", "\nSERVICE", url)
+}
 
-
+pagination_sort_col <- function(x) {
+  cols <- bcdc_describe_feature(x)[["col_name"]]
+  # use OBJECTID or OBJECT_ID as default sort key, if present
+  # if not present (several GSR tables), use SEQUENCE_ID
+  # Then try FEATURE_ID
+  for (idcol in c("OBJECTID", "OBJECT_ID", "SEQUENCE_ID", "FEATURE_ID")) {
+    if (idcol %in% cols) return(idcol)
+  }
+  #Otherwise use the first column - this is likely pretty fragile
+  warning("Unable to find a suitable column to sort on for pagination. Using",
+          " the first column (", cols[1],
+          "). Please check your data for obvious duplicated or missing rows.",
+          call. = FALSE)
+  cols[1]
```

Comment:
What if we did pass a logical through this function (TRUE for OBJECT_ID etc col, FALSE for the first col option) which then is passed to a conditional that checks if there are duplicated columns? Is that too edge-casey?

Reply:
I feel like for the limited number of times this will happen it will add enough slightly obscure code that it's not really worth it - I think the warning should be sufficient?

Reply:
Sure that's fine.